### PR TITLE
Use correct compare method for numbers

### PR DESCRIPTION
**Pull request by daniel-wer**
I didn't manage to change the flowtypes for the compareBy methods, so that one can only be called with numbers and the other with strings. However, I fixed our usage of compareBy and localeCompareBy in the admin views and added a console error message to warn of wrong usage in the future.

### URL of deployed dev instance (used for testing):
- https://repairsorting.webknossos.xyz

### Steps to test:
- Open the admin views that are affected by this PR and check whether the sorting works correctly.

### Issues:
- fixes incorrect sorting in some admin views

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/dashboard/gallery_dataset_view.js [hunk 1]
```diff
@@ -75,12 +75,12 @@ class GalleryDatasetView extends React.PureComponent<Props, State> {
...
         // Sort each group of datasets
-        [organization, datasets.sort(Utils.localeCompareBy(([]: DatasetType[]), "created", false))],
+        [organization, datasets.sort(Utils.compareBy(([]: DatasetType[]), "created", false))],
       )
       .value()
       .sort(
         // Sort groups by creation date of first dataset
-        Utils.localeCompareBy(
+        Utils.compareBy(
           ([]: DatasetType[]),
```
Comment (philippotto) [inline app/assets/javascripts/dashboard/gallery_dataset_view.js]: Can you change this line to `([]: Array<[string, APIDatasetType]>)`? Then, the old error should have been caught in my opinion.

Comment (philippotto) [reply app/assets/javascripts/dashboard/gallery_dataset_view.js]: I wonder why flow didn't complain here in the first place :thinking:

Comment (daniel-wer) [reply app/assets/javascripts/dashboard/gallery_dataset_view.js]: Yes I'm wondering that as well, I'll investigate :)

**Comment by daniel-wer** [comment]
> Maybe we should remove the possibility to provide the key as a string completely? So, one would have to write el => el.created instead of created. Would be a bit more (keyboard) typing, but also better (static) typing..

I like that :)

**Pull request merged**